sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkThriftServerProtocolVersionsSuite.scala

```diff
@@ -0,0 +1,208 @@
+      }
+    }
+
+    test(s"$version get bigint type") {
+      testWithProtocolVersion(version, "SELECT cast(1 as bigint)") { rs =>
+        assert(rs.next())
+        assert(rs.getLong(1) === 1L)
+      }
+    }
+
+    test(s"$version get float type") {
+      testWithProtocolVersion(version, "SELECT cast(1.2 as float)") { rs =>
+        assert(rs.next())
+        assert(rs.getFloat(1) === 1.2F)
+      }
+    }
+
+    test(s"$version get double type") {
+      testWithProtocolVersion(version, "SELECT cast(1.2 as double)") { rs =>
+        assert(rs.next())
+        assert(rs.getDouble(1) === 1.2D)
+      }
+    }
+
+    // TODO: enable this test case after SPARK-28463 and SPARK-26969
+    ignore(s"$version get decimal type") {
+      testWithProtocolVersion(version, "SELECT cast(1 as decimal(18, 2)) as c") { rs =>
+        assert(rs.next())
+        assert(rs.getBigDecimal(1) === new java.math.BigDecimal("1.00"))
+      }
+    }
+
+    test(s"$version get string type") {
+      testWithProtocolVersion(version, "SELECT 'str'") { rs =>
+        assert(rs.next())
+        assert(rs.getString(1) === "str")
+      }
+    }
+
+    test(s"$version get char type") {
+      testWithProtocolVersion(version, "SELECT cast('char-str' as char(10))") { rs =>
+        assert(rs.next())
+        assert(rs.getString(1) === "char-str")
+      }
+    }
+
+    test(s"$version get varchar type") {
+      testWithProtocolVersion(version, "SELECT cast('varchar-str' as varchar(10))") { rs =>
+        assert(rs.next())
+        assert(rs.getString(1) === "varchar-str")
+      }
+    }
+
+    // TODO: enable this test case after SPARK-28474
+    ignore(s"$version get binary type") {
+      testWithProtocolVersion(version, "SELECT cast('ABC' as binary)") { rs =>
+        assert(rs.next())
+        assert(rs.getString(1) === "ABC")
+      }
+    }
+
+    test(s"$version get boolean type") {
+      testWithProtocolVersion(version, "SELECT true") { rs =>
+        assert(rs.next())
+        assert(rs.getBoolean(1) === true)
+      }
+    }
+
+    test(s"$version get date type") {
+      testWithProtocolVersion(version, "SELECT cast('2019-07-22' as date)") { rs =>
+        assert(rs.next())
+        assert(rs.getDate(1) === Date.valueOf("2019-07-22"))
+      }
+    }
+
+    test(s"$version get timestamp type") {
+      testWithProtocolVersion(version, "SELECT cast('2019-07-22 18:14:00' as timestamp)") { rs =>
+        assert(rs.next())
+        assert(rs.getTimestamp(1) === Timestamp.valueOf("2019-07-22 18:14:00"))
+      }
+    }
+
+    // TODO: enable this test case after port HIVE-10646
+    ignore(s"$version get void") {
+      testWithProtocolVersion(version, "SELECT null") { rs =>
+        assert(rs.next())
+        assert(rs.getString(1) === null)
+      }
+    }
+
+    // We do not fully support interval type
+    ignore(s"$version get interval type") {
+      testWithProtocolVersion(version, "SELECT interval '1' year '2' day") { rs =>
+        assert(rs.next())
```

Comment:
nit: should we still assert what the returned value is equal to here?

Reply:
it will throw exception.
Client log:
```sql
0: jdbc:hive2://localhost:10000/default> SELECT interval '1' year '2' day;
Error: java.lang.IllegalArgumentException: Unrecognized type name: interval (state=,code=0)
```
Server log:
```java
java.lang.RuntimeException: java.lang.IllegalArgumentException: Unrecognized type name: interval
	at org.apache.hive.service.cli.session.HiveSessionProxy.invoke(HiveSessionProxy.java:83)
	at org.apache.hive.service.cli.session.HiveSessionProxy.access$000(HiveSessionProxy.java:36)
	at org.apache.hive.service.cli.session.HiveSessionProxy$1.run(HiveSessionProxy.java:63)
	at java.security.AccessController.doPrivileged(AccessController.java:770)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1746)
	at org.apache.hive.service.cli.session.HiveSessionProxy.invoke(HiveSessionProxy.java:59)
	at com.sun.proxy.$Proxy26.getResultSetMetadata(Unknown Source)
	at org.apache.hive.service.cli.CLIService.getResultSetMetadata(CLIService.java:436)
	at org.apache.hive.service.cli.thrift.ThriftCLIService.GetResultSetMetadata(ThriftCLIService.java:607)
	at org.apache.hive.service.cli.thrift.TCLIService$Processor$GetResultSetMetadata.getResult(TCLIService.java:1533)
	at org.apache.hive.service.cli.thrift.TCLIService$Processor$GetResultSetMetadata.getResult(TCLIService.java:1518)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
	at org.apache.hive.service.auth.TSetIpAddressProcessor.process(TSetIpAddressProcessor.java:53)
	at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:310)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:819)
Caused by: java.lang.IllegalArgumentException: Unrecognized type name: interval
	at org.apache.hive.service.cli.Type.getType(Type.java:169)
	at org.apache.hive.service.cli.TypeDescriptor.<init>(TypeDescriptor.java:53)
	at org.apache.hive.service.cli.ColumnDescriptor.<init>(ColumnDescriptor.java:53)
	at org.apache.hive.service.cli.TableSchema.<init>(TableSchema.java:52)
	at org.apache.spark.sql.hive.thriftserver.SparkExecuteStatementOperation$.getTableSchema(SparkExecuteStatementOperation.scala:314)
	at org.apache.spark.sql.hive.thriftserver.SparkExecuteStatementOperation.resultSchema$lzycompute(SparkExecuteStatementOperation.scala:69)
	at org.apache.spark.sql.hive.thriftserver.SparkExecuteStatementOperation.resultSchema(SparkExecuteStatementOperation.scala:64)
	at org.apache.spark.sql.hive.thriftserver.SparkExecuteStatementOperation.getResultSetSchema(SparkExecuteStatementOperation.scala:158)
	at org.apache.hive.service.cli.operation.OperationManager.getOperationResultSetSchema(OperationManager.java:209)
	at org.apache.hive.service.cli.session.HiveSessionImpl.getResultSetMetadata(HiveSessionImpl.java:773)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.hive.service.cli.session.HiveSessionProxy.invoke(HiveSessionProxy.java:78)
	... 18 more
```

Reply:
Would it fix it if it was converted to string like other unsupported types (Arrays, Maps etc.) in https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala#L107 ?